### ISSUE TITLE
False positive in `only_todo`

### ISSUE THREAD

**Issue by mdeaconu** [realm/SwiftLint#3324]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

When the code has a comment that starts with `DEBUG`, false positive `only_todo` violations are reported. If the word is not the first one then everything is fine.

An example is put at the end of description.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths 
Linting 'test.swift' (1/1)
/Users/mihai.deaconu/test/test.swift:9:1: warning: Use TODO/FIXME exclusively Violation: This makes our code base easier to search. (only_todo)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
```bash
$ swiftlint version
0.39.2
```
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
`Homebrew`
* Paste your configuration file:

```yml
custom_rules:
   comments_space: # From https://github.com/brandenr/swiftlintconfig
     name: "Space After Comment"
     regex: '(^[ \t]*[/]{2,3}\w+)'
     message: "There should be a space after // or ///"
     severity: warning
   only_todo:
     name: "Use TODO/FIXME exclusively"
     regex: '^[ \t]*[/]{2,3}\s*(?!TODO|FIXME|MARK|WTF|NOTE|WARNING|NOOP|XML|JSON|UTF)[A-Z!?]{3,}($|\s|\b)'
     message: "This makes our code base easier to search."
     severity: warning
   no_should:
     name: "Avoid should in tests"
     regex: 'it\("should'
     message: "State what it does, not what it should do."
     severity: warning
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
 No.
* Which Xcode version are you using (check `xcodebuild -version`)?
```
Xcode 11.6
Build version 11E708
```
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
import Foundation

enum Test: String {
    case one = "One"
    case two = "Two"

    static var test: Test {
        #if DEBUG
        // DEBUG comment
        return .one
        #else
        return .two
        #endif
     }
 }
```

**Comment by mdeaconu** [comment]
Sorry for the noise. Wrong configuration on my side. 🙏
